### PR TITLE
[ACR] `az acr agentpool`: Fix delete polling

### PR DESCRIPTION
**Pull request by nguyenm2151**
**Related command**
az acr agentpool delete

**Description**
Revert some changes from this PR https://github.com/Azure/azure-cli/pull/30195 since we found an regression. Found an issue with self.cmd not detecting a command failure in this case. https://github.com/Azure/azure-cli/issues/30478

**Testing Guide**
acr agentpool delete -n {agents_name} -r {registry_name} -y

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by nguyenm2151** [comment]
@zhoxing-ms @yonzhan can I get a review for this

File: src/azure-cli/azure/cli/command_modules/acr/tests/latest/test_acr_agentpool_commands.py [hunk 1]
```diff
@@ -3,11 +3,11 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
-from azure.cli.testsdk import ScenarioTest, StorageAccountPreparer, ResourceGroupPreparer, record_only
-
+from azure.cli.testsdk import ScenarioTest, StorageAccountPreparer, ResourceGroupPreparer, record_only, live_only
 
 class AcrAgentPoolCommandsTests(ScenarioTest):
 
+    @live_only()
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/acr/tests/latest/test_acr_agentpool_commands.py:10]: Can you please add a comment to explain why need to add `@live_only()` to this test?

**Comment by yanzhudd** [comment]
May I ask if this change would be perceptible to users?

**Comment by nguyenm2151** [comment]
Currently they see an error message after running that command (hence the regression). This change makes the command returns with no error message. So yes users will be able to see it

**Comment by nguyenm2151** [comment]
@yanzhudd i have changed the PR title since this change is perceptible to users

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**